test/Test/Database/LSMTree/ModelIO/Class.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE LambdaCase     #-}
+{-# LANGUAGE TypeFamilies   #-}
+module Test.Database.LSMTree.ModelIO.Class (
+    LookupResult (..)
+  , TableHandle (..)
+  ) where
+
+import           Data.Bifunctor (second)
+import           Data.Kind (Constraint, Type)
+import           Database.LSMTree.Common (IOLike)
+import           Database.LSMTree.Model.Normal (Update (..))
+import qualified Database.LSMTree.ModelIO.Normal as M
+import qualified Database.LSMTree.Normal as R
+
+-- | Result of a single point lookup.
+--
+-- TODO: should this be shared?
```

Comment:
I'm asking myself the same question while working on #4. I think we should share this, at least for the tests, and maybe between the model and implementation as long as we can hide the abstraction in the API.